insert.go

```diff
@@ -30,6 +30,50 @@ func (db *DB) InsertRaw(stream string, ts time.Time, dims bytemap.ByteMap, vals
 		return fmt.Errorf("No wal found for stream %v", stream)
 	}
 
+	// Write separate rows for array values if necessary
+	var lastErr error
+	mainVals := bytemap.Build(func(include func(string, interface{})) {
+		vals.IterateValues(func(key string, value interface{}) bool {
+			switch v := value.(type) {
+			case float64:
+				include(key, v)
+			case int:
+				include(key, float64(v))
+			case []float64:
+				// include first value with main vals
+				include(key, v[0])
+				// do separate inserts for additional values
+				for i := 1; i < len(v); i++ {
+					subVals := bytemap.Build(func(subInclude func(string, interface{})) {
+						subInclude(key, v[i])
+					}, nil, true)
+					if subErr := db.doInsertRaw(w, ts, dims, subVals); subErr != nil {
+						lastErr = subErr
+					}
+				}
+			case []int:
+				// include first value with main vals
+				include(key, float64(v[0]))
+				// do separate inserts for additional values
+				for i := 1; i < len(v); i++ {
+					subVals := bytemap.Build(func(subInclude func(string, interface{})) {
+						subInclude(key, float64(v[i]))
+					}, nil, true)
+					if subErr := db.doInsertRaw(w, ts, dims, subVals); subErr != nil {
+						lastErr = subErr
+					}
+				}
+			}
+			return true
```

Comment:
Should there be a default case here?  If I follow the logic correctly, calling `Insert` with say a string would return a `nil` error, leading the caller to believe that the insert was successful.

Reply:
The only supported values are numeric, which in practice is ints, float64s or arrays thereof. I'll Add a default case to log an error if we get something else.

Reply:
Cool, thanks!  Just finished looking over the rest and everything else looks good.

Reply:
Fixed.